circuits/transfer/src/execute/input/witness.rs

```diff
@@ -0,0 +1,60 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use dusk_poseidon::cipher::PoseidonCipher;
+
+use dusk_plonk::prelude::*;
+
+/// Coupled code
+///
+/// Currently, Plonk is not a dependency of phoenix-core. This means the circuit
+/// construction of the note must be done here.
+///
+/// Ideally, there would be a `fn hash_inputs_witness(&self, composer)`
+/// implemented for `Note`.
+///
+/// Since the circuit will perform a pre-image check over the result of this
+/// function, the structure is safe
+///
+/// However, if `Note::hash_inputs` ever change, this circuit will be broken
```

Comment:
It shouldn't be a breaking change. We could have some strategy in the future to take the hasher as generic and return some digest. This is pretty much common in crypto libs

Reply:
Even if we take the hasher as generic, a change in the order `Note` is hashed it would still be breaking change, since the hash would differ between versions for the same `Note` instance. I do get your point though, and it's nice to have the comment in there to explain it.